From 4223d55dcb50fbbfa44d06934010f7d6ba255c3b Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 02:16:26 -0400
Subject: [pci] Add bridge DMA alias quirk for ITE bridge

Message-id: <20140905021626.32574.47762.stgit@gir.stowe>
Patchwork-id: 91367
O-Subject: [RHEL7.1 PATCH 08/17] PCI: Add bridge DMA alias quirk for ITE bridge
Bugzilla: 1136537
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1136537
Upstream Status: 98ca50db1bb91822fc190d3ebec0f80c556c8cfb

commit 98ca50db1bb91822fc190d3ebec0f80c556c8cfb
Author: Alex Williamson <alex.williamson@redhat.com>
Date:   Mon Jun 9 12:43:25 2014 -0600

    PCI: Add bridge DMA alias quirk for ITE bridge

    The ITE 8892 is a PCIe-to-PCI bridge but doesn't have a PCIe capability.
    Quirk it so we can figure out the DMA alias for devices below the bridge,
    so they work correctly with an IOMMU.

    [bhelgaas: add changelog]
    Link: https://bugzilla.kernel.org/show_bug.cgi?id=73551
    Reported-by: Ronald <rwarsow@gmx.de>
    Tested-by: Ronald <rwarsow@gmx.de>
    Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 5170642..1e7253b 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -3413,6 +3413,8 @@ DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_ASMEDIA, 0x1080,
     quirk_use_pcie_bridge_dma_alias);
 /* Tundra 8113, https://bugzilla.kernel.org/show_bug.cgi?id=44881#c43 */
 DECLARE_PCI_FIXUP_HEADER(0x10e3, 0x8113, quirk_use_pcie_bridge_dma_alias);
+/* ITE 8892, https://bugzilla.kernel.org/show_bug.cgi?id=73551 */
+DECLARE_PCI_FIXUP_HEADER(0x1283, 0x8892, quirk_use_pcie_bridge_dma_alias);
 
 static struct pci_dev *pci_func_0_dma_source(struct pci_dev *dev)
 {
-- 
1.7.1